### ISSUE TITLE
1.5 Hours Course

### ISSUE THREAD

**Issue by anish-kum** [anish-kum/PPP-Spark-2025-2026#10]
90 minute PPP

**Comment by anish-kum** [comment]
<img width="334" height="328" alt="Image" src="https://github.com/user-attachments/assets/05582cad-e190-4e9b-9959-8e24e69ba4ff" />

<img width="298" height="312" alt="Image" src="https://github.com/user-attachments/assets/dc9255a0-bd32-48d5-b464-5c2abec0e84d" />

<img width="324" height="352" alt="Image" src="https://github.com/user-attachments/assets/169259c0-b891-4d28-9351-8c0f78bc80ae" />

What I was learning:

**_For the Talking Pytorch with Soumith Chintala section:_**

- Co-founder of PyTorch: Soumith Chintala discusses his role in developing PyTorch and its evolution as a leading deep learning framework.

- Community Focus: Emphasizes the importance of community feedback in shaping PyTorch's features and usability.

- Dynamic Computation Graphs: Highlights PyTorch's dynamic computation graph, which allows for more intuitive model building and debugging.

- Research and Production: Discusses how PyTorch bridges the gap between research and production, making it easier for researchers to deploy models.

- Future Directions: Soumith shares insights on future developments and enhancements planned for PyTorch.

**_For the Introduction to Pytorch section from 4.1 to 4.5_**

- Tensors: The fundamental data structure in PyTorch, similar to NumPy arrays but with GPU support for accelerated computing.

- Creating Tensors: Various methods to create tensors, including from lists, NumPy arrays, and using built-in functions like torch.tensor().

- Tensor Operations: Basic operations such as addition, multiplication, and reshaping.

- Autograd: Introduces automatic differentiation, allowing for the computation of gradients for tensor operations, essential for training neural networks.

- Gradient Tracking: Explains how to enable and disable gradient tracking for tensors, optimizing memory usage during model training.


Questions the content didn't cover that I'd like an answer to:

- When should I use a NumPy array versus a PyTorch tensor in practice?

- Why is gradient tracking important, and what happens if I forget to enable or disable it?


Note: Still deciding on when to transition to another project (which is tentative and I very well may not since actually coding with Pytorch compared to the theory I've been learning this whole time is more interesting now, so we'll see).

**Comment by drewmca-pkwy** [comment]
@anish-kum Good to hear you're finding the coding part of the experience better than the theory, Anish. So long as the content is engaging and you feel like you're making progress, staying the course makes sense.
